CreeDictionary/search_quality/combine_samples.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+"""
+Combine samples in survey_results directory into one
+
+The algorithm is:
+
+ 1. For each word list:
+     1. For each query term, assign 1/n points to the n’th result.
+         For example, if the list has "foo: bar, baz, qux", then "bar" gets 1
+         point, "baz" gets 1/2 point, and "qux" gets 1/3 point as results for
+         "foo"
+ 2. For each query term:
+      1. Add up all the points for all the results across all lists
+      2. Sort by highest points first
+"""
+import csv
+from argparse import ArgumentParser, BooleanOptionalAction
+from collections import defaultdict
+from io import StringIO
+from pathlib import Path
+
+
+def iter_results(input_file):
+    """Yield dicts for rows from a CSV file, whether it has a header or not
+
+    The reason we don’t just add a header to every .csv file and use DictReader
+    is that these .csv files may have arbitrarily many columns.
+    """
+    reader = csv.reader(input_file)
+
+    have_header = None
+    column_count = None
+    have_notes = False
+
+    seen_queries = set()
+
+    first_row = True
+    for row_index, row in enumerate(reader):
+        print("got row", row)
+        if first_row:
+            first_row = False
+
+            have_header = row[0].lower().strip() == "query"
+            if have_header:
+                column_count = len(row)
+            if have_header and row[-1].lower().strip() == "notes":
+                if row[-2].strip():
+                    raise Exception("Must have blank column before notes column")
+                have_notes = True
+
+            if have_header:
+                # Don’t yield first line
+                continue
+
+        if have_header:
+            if len(row) > column_count:
+                raise Exception(f"More entries in row {row} than header values")
+
+        query_term = row[0].strip()
+        max_col = column_count
```

Comment:
I think it's possible that `column_count` is still set to `None` here

Reply:
Yup, that is intentional, but I thank you for asking about that because its usage is a bit obscure. I have added a comment, and am happy to revisit if you’d like the code cleaned up further.